.github/workflows/publish.yml

```diff
@@ -0,0 +1,179 @@
+name: Publish Builds
+
+# manual run only for publishing
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        required: true
+        description: "Release tag to publish. (i.e: v1.4, v1.4.1, etc)"
+        default: "vx.y.z"
+      draftRelease:
+        description: "Is this a 'draft'/'*-beta' release? (MUST ANSWER EITHER 'true' OR 'false' AS STRING INPUT)"
+        required: true
+        default: "false"
+      preRelease:
+        description: "Is this a 'prerelease'/'*-rc'? (MUST ANSWER EITHER 'true' OR 'false' AS STRING INPUT)"
+        required: true
+        default: "false"
+    tags:
+      - v[0-9].[0-9]**
+      - v[0-9].[0-9]**-**
+
+jobs:
+  taisei-release:
+    name: Release Actions
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'workflow_dispatch' }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          ref: refs/tags/${{ github.event.inputs.tag }}
+
+      - name: Set Env Vars
+        run: |
+          echo "taisei_version=$(echo '${{ github.event.inputs.tag }}' | sed 's/^v//g')" >> $GITHUB_ENV
+          # adds default changelog, which is blank, in case you forget so it doesn't hold up the whole release
+          # edit it on the actual GH release and commit what you write to master afterwards for posterity
+          if [ ! -f "changelogs/changelog-${{ github.event.inputs.tag }}.md" ]; then
+            echo "taisei_changelog_tag=default" >> $GITHUB_ENV
+          else
+            echo "taisei_changelog_tag=${{ github.event.inputs.tag }}" >> $GITHUB_ENV
+          fi
```

Comment:
We can always pass in a blank changelog and then change the release "in post", so to speak. The idea with this is that if there _is_ a changelog we've written ahead of time, it can get used, but otherwise, it'll just fill it in with a blank one we can edit later. It's up to you, though.